### PR TITLE
Improve music view

### PR DESCRIPTION
**Pull request by sigvef**
This adds bpm-based lines to the music view, and some additional music
variable info panels.

image

### CODE REVIEW

File: nin/frontend/app/scripts/directives/waveform.js [hunk 1]
```diff
@@ -1,19 +1,54 @@
 const Waveform = require('../../lib/waveform');
 
 function waveform($window) {
+
```
Comment (aleksanb) [inline nin/frontend/app/scripts/directives/waveform.js]: Ref style discussion, i prefer no newline after function declarations.
How about the rest of you?

File: nin/frontend/app/scripts/directives/waveform.js [hunk 2]
```diff
@@ -1,19 +1,54 @@
...
+  const white = '#d3dde3';
+  const grey = 'rgb(42, 62, 72)';
+  const pixelSize = 1 / window.devicePixelRatio;
+  const beatOffset = (PROJECT.music.beatOffset || 0);
+  console.log(PROJECT.music);
```
Comment (aleksanb) [inline nin/frontend/app/scripts/directives/waveform.js]: log

File: nin/frontend/app/scripts/directives/waveform.js [hunk 3]
```diff
@@ -1,19 +1,54 @@
...
+    const backgroundColor = i % 32 >= 16 ? backgroundColorA : backgroundColorB;
+    if(i % 4 == 0) {
+      gradientArguments.push(`${lineColor} {{beatLength * ${i}}}px`);
+      gradientArguments.push(`${lineColor} {{beatLength * ${i} + ${pixelSize}}}px`);
+    }
+    gradientArguments.push(`${backgroundColor} {{beatLength * ${i} + ${pixelSize}}}px`);
+    gradientArguments.push(`${backgroundColor} {{beatLength * ${i + 1}}}px`);
+  }
+
+  console.log(gradientArguments);
```
Comment (aleksanb) [inline nin/frontend/app/scripts/directives/waveform.js]: more logs

**Comment by sigvef** [comment]
Superceded by https://github.com/ninjadev/nin/pull/407 .

**Pull request closed without merging**